packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -2897,30 +2885,114 @@ class StackViewport extends Viewport implements IStackViewport, IImagesLoader {
   public isReferenceViewable(
     viewRef: ViewReference,
     options: ReferenceCompatibleOptions = {}
-  ): boolean {
+  ): boolean | unknown {
     if (!super.isReferenceViewable(viewRef, options)) {
       return false;
     }
 
-    let { imageURI } = options;
     const { referencedImageId, sliceIndex } = viewRef;
 
     if (viewRef.volumeId && !referencedImageId) {
       return options.asVolume === true;
     }
 
     let testIndex = this.getCurrentImageIdIndex();
+    const currentImageId = this.imageIds[testIndex];
+
     if (options.withNavigation && typeof sliceIndex === 'number') {
       testIndex = sliceIndex;
     }
-    const imageId = this.imageIds[testIndex];
-    if (!imageId) {
+    if (!currentImageId) {
       return false;
     }
+
+    /**
+     * For overlay, we don't care about the imageId. We just want to
+     * know if the viewport can show the overlay image. We return true
+     * if the viewport has the same orientation as the overlay image.
+     * We may need to consider if they are in the same bounds, but
+     * that's a consideration for later.
+     */
+    if (options.asOverlay && referencedImageId) {
```

Comment:
This makes the function a little too complex - can you extract the asOverlay handling into it's own method which returns a targetId type value?